### PR TITLE
Add the Profile Screen

### PR DESCRIPTION
**Pull request by estacioneto**

### CODE REVIEW

File: client/components/ProfileForm.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  }
+
+  maskPhoneNumber = text => {
+    return (
+      text &&
+      `${text.length > 0 ? '(' : ''}${text.substr(0, 2)}${
+        text.length > 2 ? ') ' : ''
+      }${text.substr(2, 1)}${text.length > 3 ? ' ' : ''}${text.substr(3, 4)}${
+        text.length > 7 ? '-' : ''
+      }${text.substr(7, 4)}`
```
Comment (lucasecdb) [inline client/components/ProfileForm.js]: regex?

Comment (estacioneto) [reply client/components/ProfileForm.js]: How?

File: client/components/ProfileForm.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  updateField = field => text =>
+    this.setState({
+      profile: {
+        ...this.state.profile,
+        [field]: text,
+      },
+    })
+
+  areEqual = (
```
Comment (lucasecdb) [inline client/components/ProfileForm.js]: `compare`? ⛏

File: client/components/screens/Profile.js [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    navigation: PropTypes.object,
+    profile: PropTypes.object,
+    loading: PropTypes.bool,
+    updateProfile: PropTypes.func,
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
```
Comment (lucasecdb) [inline client/components/screens/Profile.js]: why don't you use class properties?

File: client/components/screens/Profile.js [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+  saveProfile = profile => {
+    const {
+      profile: { uid },
+    } = this.props
+
+    this.props.updateProfile(uid, profile)
+  }
+
+  areEqual = (
```
Comment (lucasecdb) [inline client/components/screens/Profile.js]: isn't this function duplicated? we could use ramda

File: client/package.json [hunk 5]
```diff
@@ -35,6 +35,7 @@
     "react": "16.3.1",
     "react-native": "https://github.com/expo/react-native/archive/sdk-30.0.0.tar.gz",
     "react-native-elements": "^0.19.1",
+    "react-native-masked-text": "^1.9.2",
```
Comment (lucasecdb) [inline client/package.json]: unused?

File: client/sagas/user.js [hunk 6]
```diff
@@ -31,6 +44,23 @@ const createUserDocument = user =>
...
 
+const updateUser = (uid, user) =>
+  firestore()
+    .collection('users')
+    .doc(uid)
+    .update(user)
+
+export function* userFromFirebaseSaga({ user }) {
+  const firebaseUser = yield call(getUser, user.uid)
+  if (firebaseUser.photoURL !== user.photoURL) {
```
Comment (lucasecdb) [inline client/sagas/user.js]: 🤔

File: client/components/screens/Profile.js (hunk 3 above)
Comment (estacioneto) [reply client/components/screens/Profile.js]: I needed to initialize it based on props before...

File: client/package.json (hunk 5 above)
Comment (estacioneto) [reply client/package.json]: Removing...

File: client/components/screens/Profile.js (hunk 3 above)
Comment (lucasecdb) [reply client/components/screens/Profile.js]: you can access the props on class properties e.g.
```js
state = {
  myState: this.props.myProp,
}
```

File: client/components/ProfileForm.js [hunk 7]
```diff
@@ -1,4 +1,5 @@
 import PropTypes from 'prop-types'
+import { eq } from 'lodash'
```
Comment (lucasecdb) [inline client/components/ProfileForm.js]: 😡

File: client/components/ProfileForm.js [hunk 8]
```diff
@@ -14,13 +15,8 @@ export default class ProfileForm extends Component {
...
     super(props)
 
-    const { displayName, email, phoneNumber } = props.profile
     this.state = {
-      profile: {
-        displayName,
-        email,
-        phoneNumber,
-      },
+      profile: props.profile,
```
Comment (lucasecdb) [inline client/components/ProfileForm.js]: you can move this to a class property

**Pull request merged**